File: TripKit/Classes/helpers/classification/TripRequest+Classify.m
//
//  TripRequest+Classify.m
//  TripKit
//
//  Created by Adrian Schoenig on 11/06/2015.
//  Copyright (c) 2015 SkedGo Pty Ltd. All rights reserved.
//

#import "TripRequest+Classify.h"

#import "TKTripClassifier.h"

@implementation TripRequest (Classify)

- (void)updateTripGroupClassificationsUsingClassifier:(nonnull id<TKTripClassifier>)classifier
{
  NSSet *tripGroups = [self tripGroups];
  [classifier prepareForClassifictionOfTripGroups:tripGroups];
  for (TripGroup *group in tripGroups) {
    NSString *newClassification = [classifier classificationOfTripGroup:group];
    NSString *oldClassification = group.classification;
    if (!oldClassification || ! [newClassification isEqual:oldClassification]) {
      group.classification = newClassification;
    }
  }
}

@end

File: TripKit/Classes/helpers/classification/TripRequest+Classify.swift
//
//  TripRequest+Classify.swift
//  TripKit-iOS
//
//  Created by Adrian Schönig on 6/8/21.
//  Copyright © 2021 SkedGo Pty Ltd. All rights reserved.
//

import Foundation

extension TripRequest {
  public func updateTripGroupClassifications(using classifier: TKTripClassifier) {
    guard let groups = self.tripGroups else { return }
    classifier.prepareForClassification(of: groups)
    for group in groups {
      let newClassification = classifier.classification(of: group)
      let oldClassification = group.classification
      if oldClassification == nil || newClassification != oldClassification {
        group.classification = newClassification
      }
    }
  }
}
